currency-rates/integrations_tests/currency-rates/handler_test.go

```diff
@@ -0,0 +1,107 @@
+//go:build integration
+
+package currencyrates
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"go.uber.org/zap"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+
+	"github.com/RinaDish/currency-rates/integrations_tests/pkg/testdb"
+	"github.com/RinaDish/currency-rates/internal/handlers"
+	"github.com/RinaDish/currency-rates/internal/repo"
+	"github.com/stretchr/testify/require"
+)
+
+const email = "testemail@gmail.com"
+
+func setupUserHandler(db *gorm.DB) (handlers.SubscribeHandler, *repo.Repository) {
+	logger := zap.NewNop().Sugar()
+
+	adminRepository := repo.NewAdminRepository(db, logger)
+
+	handler := handlers.NewSubscribeHandler(logger, adminRepository)
+
+	return handler, adminRepository
+}
+
+func sendMail(email string, db *gorm.DB) *http.Response {
+	userHandler, _ := setupUserHandler(db)
+	form := url.Values{}
+	form.Add("email", email)
+
+	formData := form.Encode()
+
+	req := httptest.NewRequest(http.MethodPost, "/subscribe", strings.NewReader(formData))
+
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	w := httptest.NewRecorder()
+	userHandler.CreateSubscription(w, req)
+
+	response := w.Result()
+	defer response.Body.Close()
+
+	return response
+}
+
+func TestUserHandler(main *testing.T) {
```

Comment:
If there are pretty the same flow of test setup, you can use table-driven tests ([go.wiki article](https://go.dev/wiki/TableDrivenTests)) to decrease amount of code duplication.

Reply:
Thanks for your recommendation, but I don't prefer this option)